collection.go

```diff
@@ -117,3 +117,17 @@ func (c Collection[V]) Last() V {
 
 	return c.values[c.keys[len(c.keys)-1]]
 }
+
+func (c Collection[V]) Merge(other Collection[V]) (Collection[V], error) {
+	newCollection := makeCollection[V](0)
+
+	c.Each(func(_ any, v V) {
+		newCollection.Push(v)
+	})
+
+	other.Each(func(_ any, v V) {
+		newCollection.Push(v)
+	})
```

Comment:
You can reduce the iterations here by iterating on the keys slice of the collection with the greatest length. This would cut the iterations up 50%, depending on the size difference of both. Then, you can use the keys to index the underlying map and take the values. Something like:
```go
for i := 0; i < len(bigger); i++ {
    // use i to access the keys slice and the resulting key to take the values from the underlying map...
}
```

It'd also be interesting to use `copy` on both c.keys slices. You do, however, have to take care of duplicate keys and ensure the resulting slice contains only unique keys without wasting allocations. This is tricky, however, because we don't wasn't to waste allocations by copying both slices, nor generating too much reslicing by starting at 0 capacity.
Maybe a good approach here is to allocate a slice with a cap equal to the length of the bigger, then copy the bigger, and reslicing from there. WDYT?